athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/ElasticsearchRecordHandler.java

```diff
@@ -133,14 +137,26 @@ protected void readWithConstraint(BlockSpiller spiller, ReadRecordsRequest recor
                                       QueryStatusChecker queryStatusChecker)
             throws RuntimeException
     {
-        logger.info("readWithConstraint - enter - Domain: {}, Index: {}, Mapping: {}",
-                recordsRequest.getTableName().getSchemaName(), recordsRequest.getTableName().getTableName(),
-                recordsRequest.getSchema());
+        String domain;
+        QueryBuilder query;
+        String index;
+        if (recordsRequest.getConstraints().isQueryPassThrough()) {
+            Map<String, String> qptArgs = recordsRequest.getConstraints().getQueryPassthroughArguments();
+            domain = qptArgs.get(ElasticsearchQueryPassthrough.SCHEMA);
+            index = qptArgs.get(ElasticsearchQueryPassthrough.INDEX);
+            query = qptArgs.get(ElasticsearchQueryPassthrough.QUERY).isEmpty() ? QueryBuilders.matchAllQuery() : QueryBuilders.wrapperQuery(qptArgs.get(ElasticsearchQueryPassthrough.QUERY));
```

Comment:
if you run the verify; this will fail; so you'll probably need to override it in the `ElasticsearchMetadatahandler` because of the problem we've discussed online where the verify will want to make sure that the arguments not empty. But we still need to perform verify.

Reply:
agree, added verify check and removed isempty check as api is not accepting null or empty. for select * user needs to pass match all.